pgmpy/inference/Sampling.py

```diff
@@ -132,17 +142,30 @@ def rejection_sample(self, evidence=None, size=1):
         """
         if evidence is None:
             return self.forward_sample(size)
-        sampled = DataFrame(columns=self.topological_order)
+        types = [(var_name, 'int') for var_name in self.topological_order]
+        sampled = np.zeros(0, dtype=types).view(np.recarray)
         prob = 1
-        while len(sampled) < size:
-            _size = int(((size - len(sampled)) / prob) * 1.5)
+        i = 0
+        while i < size:
+            _size = int(((size - i) / prob) * 1.5)
             _sampled = self.forward_sample(_size)
+
             for evid in evidence:
-                _sampled = _sampled[_sampled.ix[:, evid[0]] == evid[1]]
+                _sampled = _sampled[_sampled[evid[0]] == evid[1]]
+
             prob = max(len(_sampled) / _size, 0.01)
-            sampled = sampled.append(_sampled)
-        sampled.reset_index(inplace=True, drop=True)
-        return sampled[:size]
+
+            if HAS_PANDAS is True:
```

Comment:
@khalibartan @yashu-seth Can we add an extra parameter to these methods for the kind of output that we want. So, the default value would be `None` and in that case it should work as it is now (checking if user has pandas or not and then work accordingly). But it the parameter is specified for either `recarray` or `dataframe` it will try to return that object type. I don't think this will add too much overhead and will help us in dealing with situations like this. What do you guys think ?